kategory/src/main/kotlin/kategory/data/StateT.kt

```diff
@@ -16,8 +16,8 @@ class StateT<F, S, A>(
     class F private constructor()
 
     companion object {
-        inline operator fun <reified F, S, A> invoke(run: StateTFunKind<F, S, A>, MF: Monad<F> = monad<F>()): StateT<F, S, A> =
-                StateT(MF, run)
+        operator fun <F, S, A> invoke(run: (S) -> HK<F, Tuple2<S, A>>, MF: Monad<F>): StateT<F, S, A> =
```

Comment:
This one was tweaked by @raulraja because I had problems to invoke the StateT from `StateTInstances`. Feels like the type F cannot be reified here, otherwise it wouldn't be able to be used from the overriden `get()` and `set` methods inside `StateTInstances` because of:

![captura de pantalla 2017-07-26 a las 8 42 48](https://user-images.githubusercontent.com/6547526/28607945-7aae92c4-71de-11e7-8f4b-6a510b38df81.png)